File: superstitioMain/java/superstitio/cards/general/PowerCard/drawAndEnergy/HideInPussy.java
package superstitio.cards.general.PowerCard.drawAndEnergy;

import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import superstitio.DataManager;
import superstitio.cards.SuperstitioCard;
import superstitio.cards.general.GeneralCard;
import superstitio.powers.EasyBuildAbstractPowerForPowerCard;
import superstitioapi.actions.ChoseCardFromHandCardSelectScreen;


public class HideInPussy extends GeneralCard {
    public static final String ID = DataManager.MakeTextID(HideInPussy.class);

    public static final CardType CARD_TYPE = CardType.POWER;

    public static final CardRarity CARD_RARITY = CardRarity.RARE;

    public static final CardTarget CARD_TARGET = CardTarget.SELF;

    private static final int COST = 2;
    private static final int COST_UPGRADED_NEW = 1;
    private static final int MAGIC = 1;

    public HideInPussy() {
        super(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET);
        this.setupMagicNumber(MAGIC);
    }

    @Override
    public void use(AbstractPlayer player, AbstractMonster monster) {
        addToBot_applyPower(new HideInPussyPower(this.magicNumber));
    }

    @Override
    public void upgradeAuto() {
        upgradeBaseCost(COST_UPGRADED_NEW);
    }

    public static class HideInPussyPower extends EasyBuildAbstractPowerForPowerCard {
        public HideInPussyPower(int amount) {
            super(amount);
        }

        @Override
        public void atEndOfTurn(boolean isPlayer) {
            if (!isPlayer || AbstractDungeon.player.hand.isEmpty()) return;
            new ChoseCardFromHandCardSelectScreen(card -> {
                card.freeToPlayOnce = true;
                card.retain = true;
            })
                    .setWindowText(String.format(powerCard.cardStrings.getEXTENDED_DESCRIPTION()[0], this.amount))
                    .setChoiceAmount(this.amount)
                    .setAnyNumber(true)
                    .setCanPickZero(true)
                    .addToBot();
        }

        @Override
        public void updateDescriptionArgs() {
            setDescriptionArgs(amount);
        }

        @Override
        protected SuperstitioCard makePowerCard() {
            return new HideInPussy();
        }
    }
}

File: superstitioMain/java/superstitio/cards/general/PowerCard/drawAndEnergy/HideInPussy.kt
package superstitio.cards.general.PowerCard.drawAndEnergy

import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.monsters.AbstractMonster
import superstitio.DataManager
import superstitio.cards.SuperstitioCard
import superstitio.cards.NormalCard
import superstitio.powers.EasyBuildAbstractPowerForPowerCard
import superstitioapi.actions.ChoseCardFromHandCardSelectScreen
import superstitioapi.utils.setDescriptionArgs

class HideInPussy : NormalCard(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET)
{
    init
    {
        this.setupMagicNumber(MAGIC)
    }

    override fun use(player: AbstractPlayer?, monster: AbstractMonster?)
    {
        addToBot_applyPower(HideInPussyPower(this.magicNumber))
    }

    override fun upgradeAuto()
    {
        upgradeBaseCost(COST_UPGRADED_NEW)
    }

    class HideInPussyPower(amount: Int) : EasyBuildAbstractPowerForPowerCard(amount)
    {
        override fun atEndOfTurn(isPlayer: Boolean)
        {
            if (!isPlayer || AbstractDungeon.player.hand.isEmpty) return
            ChoseCardFromHandCardSelectScreen { card: AbstractCard ->
                card.freeToPlayOnce = true
                card.retain = true
            }
                .setWindowText(String.format(powerCard.cardStrings.getEXTENDED_DESCRIPTION(0), this.amount))
                .setChoiceAmount(this.amount)
                .setAnyNumber(true)
                .setCanPickZero(true)
                .addToBot()
        }

        override fun updateDescriptionArgs()
        {
            setDescriptionArgs(amount)
        }

        override fun makePowerCard(): SuperstitioCard
        {
            return HideInPussy()
        }
    }

    companion object
    {
        val ID: String = DataManager.MakeTextID(HideInPussy::class.java)

        val CARD_TYPE: CardType = CardType.POWER

        val CARD_RARITY: CardRarity = CardRarity.RARE

        val CARD_TARGET: CardTarget = CardTarget.SELF

        private const val COST = 2
        private const val COST_UPGRADED_NEW = 1
        private const val MAGIC = 1
    }
}

